### PR TITLE
Fix roundstart operating computers not linking properly

### PR DESCRIPTION
**Pull request by Absolucy**
## About The Pull Request

Roundstart op computers sometimes only do a "half" link, where it's linked but patient info won't appear on the operating computer. This should fix that.

## Why It's Good For The Game

bugfix

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

23-06-02-1685753881-dreamseeker

</details>

## Changelog
:cl:
fix: Fixed roundstart operating computers only partially linking to their given stasis bed/operating table.
/:cl:

### CODE REVIEW

**Review by EvilDragonfiend — CHANGES_REQUESTED** [review]
I think `initial_link()` should be called from another object after it detects an object to be paired, not letting two objects have the same but a bit different code.


Edit:
stasis bed one should be `detect_operating_computer()` and it should find an operating computer nearby, and call `lnitial_link()` of the operating computer.

File: code/game/machinery/stasis.dm [hunk 1]
```diff
@@ -38,6 +40,14 @@
 	. += "<span class='notice'>Alt-click to [stasis_enabled ? "turn off" : "turn on"] the machine.</span>"
 	. += "<span class='notice'>[src] is [op_computer ? "linked" : "<b>NOT</b> linked"] to an operating computer.</span>"
 
+/obj/machinery/stasis/proc/initial_link()
+	for(var/direction in GLOB.alldirs)
```
Comment (Archanial) [inline code/game/machinery/stasis.dm:41]: by the time this is invoked you may have already been linked.
Also applied to initial link bellow.

File: code/game/machinery/computer/Operating.dm [hunk 2]
```diff
@@ -17,6 +17,13 @@
 /obj/machinery/computer/operating/Initialize(mapload)
 	. = ..()
 	linked_techweb = SSresearch.science_tech
+	if(mapload)
+		return INITIALIZE_HINT_LATELOAD
```
Comment (PowerfulBacon) [inline code/game/machinery/computer/Operating.dm]: Doesn't late load get called instantly if not in mapload anyway?

File: code/game/machinery/computer/Operating.dm [hunk 3]
```diff
@@ -48,17 +48,23 @@
...
+				var/obj/structure/table/optable/new_table = locate(/obj/structure/table/optable) in get_step(src, direction)
+				if(new_table && !new_table.computer)
+					table = new_table
+			if(!sbed)
+				var/obj/machinery/stasis/new_bed = locate(/obj/machinery/stasis) in get_step(src, direction)
+				if(new_bed && !new_bed.op_computer)
+					sbed = new_bed
+	if(table)
+		table.computer = src
+		src.table = table
```
Comment (EvilDragonfiend) [inline code/game/machinery/computer/Operating.dm]: that reads confusing. parameter names should be different. do the same to sbed

**Review by EvilDragonfiend — APPROVED** [review]
seems fine except for the parameter names.

File: code/game/machinery/computer/Operating.dm [hunk 4]
```diff
@@ -48,24 +48,27 @@
...
+
+/obj/machinery/computer/operating/proc/find_sbed()
+	for(var/direction in GLOB.alldirs)
+		var/obj/machinery/stasis/sbed = locate(/obj/machinery/stasis) in get_step(src, direction)
+		if(sbed && !sbed.op_computer)
+			return sbed
+
+/obj/machinery/computer/operating/proc/link_with_table(obj/structure/table/optable/new_table, obj/machinery/stasis/new_sbed)
+	new_table = new_table || find_op_table()
+	new_sbed = new_sbed || find_sbed()
```
Comment (EvilDragonfiend) [inline code/game/machinery/computer/Operating.dm]: I think `find()` procs will unintendingly happen when you only want `new_table` or `new_sbed` linked to that.
`find()` proc shouldn't be triggered if it already has linked object.

**Review by EvilDragonfiend — APPROVED** [review]

**Review by PowerfulBacon — APPROVED** [review]

**Pull request merged**